apps/api/src/app/events/services/workflow-queue/workflow.queue.service.spec.ts

```diff
@@ -0,0 +1,180 @@
+  });
+
+  afterEach(async () => {
+    await workflowQueueService.gracefulShutdown();
+  });
+
+  it('should be initialised properly', async () => {
+    expect(workflowQueueService).to.be.ok;
+    expect(workflowQueueService).to.have.all.keys(
+      'DEFAULT_ATTEMPTS',
+      'bullConfig',
+      'getBackoffStrategies',
+      'queue',
+      'queueScheduler',
+      'worker'
+    );
+    expect(workflowQueueService.DEFAULT_ATTEMPTS).to.eql(3);
+    expect(workflowQueueService.queue).to.deep.include({
+      _events: {},
+      _eventsCount: 0,
+      _maxListeners: undefined,
+      name: 'standard',
+      jobsOpts: {
+        removeOnComplete: true,
+      },
+    });
+    expect(workflowQueueService.worker).to.deep.include({
+      _eventsCount: 2,
+      _maxListeners: undefined,
+      name: 'standard',
+      blockTimeout: 0,
+      waiting: true,
+      running: true,
+    });
+  });
+
+  it('should a job added to the queue be updated as completed if works right', async () => {
+    const transactionId = uuid();
+    const _environmentId = session.environment._id;
+    const _notificationId = NotificationRepository.createObjectId();
+    const _organizationId = session.organization._id;
+    const _subscriberId = subscriber._id;
+    const _templateId = template._id;
+    const _userId = session.user._id;
+
+    // Job with fake notification template and fake notification.
+    const job: Omit<JobEntity, '_id'> = {
+      identifier: 'job-to-complete',
+      payload: {},
+      overrides: {},
+      step: {
+        template: {
+          _environmentId,
+          _organizationId,
+          _creatorId: _userId,
+          type: StepTypeEnum.TRIGGER,
+          content: '',
+        } as MessageTemplateEntity,
+        _templateId,
+      },
+      transactionId,
+      _notificationId,
+      _environmentId,
+      _organizationId,
+      _userId,
+      _subscriberId,
+      status: JobStatusEnum.PENDING,
+      _templateId,
+      digest: undefined,
+      type: StepTypeEnum.TRIGGER,
+      providerId: 'sendgrid',
+    };
+
+    const jobCreated = await jobRepository.create(job);
+
+    await workflowQueueService.addToQueue(jobCreated._id, jobCreated, 0);
+
+    await session.awaitRunningJobs(_templateId, false, 0);
+
+    const jobs = await jobRepository.find({ _environmentId, _organizationId });
+    expect(jobs.length).to.eql(1);
+
+    expect(jobs[0].status).to.eql(JobStatusEnum.COMPLETED);
+  });
+
+  it('should a job added to the queue be updated as failed if it fails', async () => {
+    const transactionId = uuid();
+    const _environmentId = session.environment._id;
+    const _notificationId = NotificationRepository.createObjectId();
+    const _organizationId = session.organization._id;
+    const _subscriberId = subscriber._id;
+    const _templateId = NotificationTemplateRepository.createObjectId();
+    const _userId = session.user._id;
+
+    // Job with fake notification template and fake notification.
```

Comment:
Job created on purpose to trigger a failure and the workflow queue service `failed` event listener callback.